experiments/AtmosLES/bomex.jl

```diff
@@ -0,0 +1,511 @@
+                                   bctype, t,
+                                   state₁⁻, diff₁⁻, aux₁⁻)
+When `bctype == 1` the `NoFluxBC` otherwise the specialized BOMEX BC is used
+"""
+#TODO This needs to be in sync with the new boundary condition interfaces
+function atmos_boundary_flux_diffusive!(nf::CentralNumericalFluxDiffusive,
+                                        bc::BOMEX_BC, 
+                                        atmos::AtmosModel, F,
+                                        state⁺, diff⁺, hyperdiff⁺, aux⁺,
+                                        n⁻,
+                                        state⁻, diff⁻, hyperdiff⁻, aux⁻,
+                                        bctype, t,
+                                        state₁⁻, diff₁⁻, aux₁⁻)
+  
+  # Floatint point precision
+  FT = eltype(state⁺)
+
+  # Establish the thermodynamic state based on the prognostic variable state
+  TS = thermo_state(atmos.moisture, atmos.orientation, state⁺, aux⁺)
+
+  # Air temperature given current thermodynamic state
+  temperature = air_temperature(TS)
+
+  # Boundary condition for bottom wall
+  if bctype != 1
+    atmos_boundary_flux_diffusive!(nf, NoFluxBC(), atmos, F,
+                                   state⁺, diff⁺, hyperdiff⁺, aux⁺, n⁻,
+                                   state⁻, diff⁻, hyperdiff⁻, aux⁻,
+                                   bctype, t,
+                                   state₁⁻, diff₁⁻, aux₁⁻)
+  else
+    # Start with the noflux BC and then build custom flux from there
+    atmos_boundary_state!(nf, NoFluxBC(), atmos,
+                          state⁺, diff⁺, aux⁺, n⁻,
+                          state⁻, diff⁻, aux⁻,
+                          bctype, t)
+
+    # Interior state velocities [u₀]
+    u₁ = state₁⁻.ρu / state₁⁻.ρ
+    # Windspeed at first interior node
+    # [Impenetrable flow, normal component of velocity is zero, horizontal components may be non-zero]
+    windspeed₁ = norm(u₁)
+    # Turbulence tensors in the interior state
+    _, τ⁻ = turbulence_tensors(atmos.turbulence, state⁻, diff⁻, aux⁻, t)
+    u_star = bc.u_star # Constant value for friction-velocity u_star == u_star
+    @inbounds begin
+      τ13⁺ = -u_star^2 * windspeed₁[1] / norm(windspeed₁) # ⟨u′w′⟩ 
+      τ23⁺ = -u_star^2 * windspeed₁[2] / norm(windspeed₁) # ⟨v′w′⟩
+      τ21⁺ = τ⁻[2,1]
+    end
+    # Momentum boundary condition [This is topography specific #FIXME: Simon's interface will improve this]
+    τ⁺ = SHermitianCompact{3, FT, 6}(SVector(0   ,
+                                             τ21⁺, τ13⁺,
+                                             0   , τ23⁺, 0))
+    # Moisture boundary condition ⟨w′qt′⟩
+    d_q_tot⁺  = SVector(0, 
+                        0, 
+                        state⁺.ρ * bc.w′qt′)
+    # Heat flux boundary condition
+    d_h_tot⁺ = SVector(0, 
+                       0, 
+                       state⁺.ρ * (bc.w′θ′ * cp_m(TS) + bc.w′qt′ * latent_heat_vapor(temperature)))
```

Comment:
I think @tapios suggests using LHF and SHF here, but it's not very clear to me. Should we prescribe constant LHF and SHF, or calculate LHF and SHF from constant w'qt' and w′θ′?

Reply:
I suggest we calculate SHF and LHF in the design doc, and use those numerical values here (i.e., not calculation in the code from w'qt' and w′θ′).

Reply:
Just to clarify, so we will have constant LHF and SHF, but then time-varying w'qt' for moisture budget (because Lv is changing with surface temperature)?

Reply:
Yes, you can view it that way (but the variations in L_v are very small).

Reply:
Sorry, never mind! I guess we should just use the prescribed surface temperature for Lv.

Reply:
Yes, that works.

Reply:
Fixed.